Sources/App/Services/Helper/QuestionnaireStorageService.swift

```diff
@@ -0,0 +1,103 @@
+//
+// This source file is part of the ENGAGE-HF-AI-Voice open source project
+//
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+import ModelsR4
+import Vapor
+
+
+/// Service for managing questionnaire data storage on disk
+@MainActor
+class QuestionnaireStorageService: Sendable {
+    typealias FilePathClosure = @Sendable (String) -> String
+    
+    private let questionnaireName: String
+    private let filePath: FilePathClosure
+    private let directoryPath: String
```

Comment:
I think we can unify the directory path and file path one element, it seems like the `filePath` is derived from the `directoryPath` and therefore can be computed from it and the phone number, I think we can therefore simply the setup and move all this logic right in here and make it an implementation detail.